Add 'RRIOT_F722' target

ws998116 opened the pull request:
We have completed development and testing for this board and would like to add the target config.

https://rotorriot.com/products/rotor-riot-brave-f7-flight-controller-made-in-usa

ws998116 commented:
One issue that we've found is in the `Receiver` tab, the `Telemetry` setting shows up twice.

![image](https://github.com/betaflight/config/assets/43047974/b1e431ab-266b-423f-9d21-88b5742338dd)

nerdCopter commented:
i'm not certain what causes dual telemetry toggles. please try nightly configurator https://github.com/betaflight/betaflight-configurator-nightlies/releases or https://app.betaflight.com in Chrome.

haslinghuis commented:
Dual telemetry shows up in case of a local build which is fixed in https://app.betaflight.com

ot0tot commented:
* Does this FC not have a LED_STRIP pin?

* Target is missing gyro interrupt pin.

* No timer for camera control pin.

Please provide the schematic in the appropriate manufacturer channel on the Betaflight Discord. Contact @sugaarK for access.

sugaarK commented:
SugarK@betaflight.com

ws998116 commented:
@ot0tot 

> * Does this FC not have a LED_STRIP pin?

No, we do not have an LED_STRIP pin.

> * Target is missing gyro interrupt pin.

I have added the gyro interrupt pin define.

> * No timer for camera control pin.

I am having trouble adding this. The camera control pin is `PC4`, and I can't seem to add a timer to this pin (`C04`) using the CLI.

haslinghuis commented:
Pin C4 does not have timers.

ws998116 commented:
> No timer for camera control pin.

> Pin C4 does not have timers.

Are we required to have a timer for the camera control pin?

haslinghuis commented:
Yes camera control seems to need a timer.

ws998116 commented:
We have updated our board to use a camera control pin with a timer, you can find the schematic in our Discord channel. The updated config file is here.

haslinghuis reviewed (approved):
Recommend using ICM42688P instead of BMI270

KarateBrot commented on configs/RRIOT_F722/config.h:
```diff
@@ -0,0 +1,98 @@
+#define MAX7456_SPI_INSTANCE SPI2
+#define FLASH_SPI_INSTANCE SPI3
+#define BARO_I2C_INSTANCE (I2CDEV_1)
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
```
You can consider using these calibration values to get voltage and current meter set up and detecting correct values. (In case they differ from the default value)
```c
#define DEFAULT_CURRENT_METER_SCALE 400
#define DEFAULT_CURRENT_METER_OFFSET 0

#define DEFAULT_VOLTAGE_METER_SCALE 110
#define DEFAULT_VOLTAGE_METER_DIVIDER 10
#define DEFAULT_VOLTAGE_METER_MULTIPLIER 1
```

ws998116 replied:
Adding these in, thanks for the suggestion!

ws998116 commented:
We would love to get this approved and move forward if possible! Please let me know if there are any additional steps that I need to take. Thank you!

nerdCopter commented:
off-subject. i added product link to description. while reviweing i noticed product page says ![image](https://github.com/user-attachments/assets/8e1e22ac-b807-474f-8bf8-e141aee08e32) without underscore.

haslinghuis commented on configs/RRIOT_F722/config.h:
```diff
@@ -0,0 +1,108 @@
+
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define DEFAULT_CURRENT_METER_SCALE 400
+#define DEFAULT_CURRENT_METER_OFFSET 0
+#define DEFAULT_VOLTAGE_METER_SCALE 110
+#define DEFAULT_VOLTAGE_METER_DIVIDER 10
+#define DEFAULT_VOLTAGE_METER_MULTIPLIER 1
```
These are all defaults. Only use when addressing 
```suggestion
```

haslinghuis commented on configs/RRIOT_F722/config.h:
```diff
@@ -0,0 +1,108 @@
+#define BEEPER_INVERTED
+
+#define PINIO1_PIN PB0
+#define PINIO1_BOX 39
```
No new line on EOF
```suggestion
#define PINIO1_BOX 39

```

haslinghuis commented on configs/RRIOT_F722/config.h:
```diff
@@ -0,0 +1,108 @@
+
+#define FC_TARGET_MCU STM32F7X2
+
+#define BOARD_NAME RRIOT_F722
```
```suggestion
#define BOARD_NAME RRIOTF722
```

ws998116 replied:
We will update the product page to match `RRIOT_F722`

ws998116 commented:
> off-subject. i added product link to description. while reviewing, i noticed product page says `RRIOTF722`.

Thanks for catching this @nerdCopter! Fixed on the product page.
